### ISSUE TITLE
[charts] browser reduce motion does not disable the initial animation in production

### ISSUE THREAD

**Issue by alexfauquette** [mui/mui-x#13477]
### Steps to reproduce

See https://github.com/mui/material-ui/pull/42537#issuecomment-2152780974

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Search keywords**: a11y animation charts

**Comment by JCQuintas** [comment]
Btw, I believe this happens because the `Global` change is triggered in about the same render step or after the animation starts. If we `skipAnimation` statically, the issue disappears.

```tsx
// docs/pages/_app.js
import { Globals } from '@react-spring/web';

Globals.assign({
  skipAnimation: true,
});
```

**Comment by Janpot** [comment]
@JCQuintas IIUC this issue is about the browser "prefers-reduced-motion" setting, not the `Globals` update. I've updated the issue description to link to the [comment](https://github.com/mui/material-ui/pull/42537#issuecomment-2152780974) in that thread. Especially the second video should illustrate the problem.

**Comment by Janpot** [comment]
> Btw, I believe this happens because the Global change is triggered in about the same render step or after the animation starts. If we skipAnimation statically, the issue disappears.

I was taking another look at the flaky regression test in core and when I set CPU throttling to 6x, I'm still seeing some sort of initial animation happening:


https://github.com/user-attachments/assets/28d6f03d-7dd8-4706-b9ce-fbf43ecef2c0

Even with global animations turned off it seems to be rendering something narrower on the initial render.

**Comment by alexfauquette** [comment]
This motion does not correspond to one of our animations. It's much closer to what happened when the responsive container resize 🤔


![image](https://github.com/user-attachments/assets/955b76f4-7319-4551-b538-8086cc0792e8)
![image](https://github.com/user-attachments/assets/f454b7b3-6d34-4d5b-9383-294db18eb306)

**Comment by Janpot** [comment]
> This motion does not correspond to one of our animations. It's much closer to what happened when the responsive container resize 🤔

By "responsive container" you still mean inside the chart, right? Because the direct parent element does not resize.

**Comment by JCQuintas** [comment]
Would it be possible that it is rendered in a size on ssr and then hydrated into another size? 🤔 

It does indeed look like the chart just changed size. Are you able to check the svg properties when this is happening?

**Comment by Janpot** [comment]
It's the core regression testing fixture. It's not SSRed, it's rendered in a nested React root created in an effect. To reproduce:
1. On the core repository
2. Run `pnpm test:regressions:dev`
3. Set CPU trhottling in the browser to 6x
4. visit http://localhost:5001/docs-getting-started-templates-dashboard-components/MainGrid.

> Are you able to check the svg properties when this is happening?

In the devtool elements panel you mean? No, not really, it always resets on reload.

**Comment by JCQuintas** [comment]
Any specific browser? I didn't see this behaviour on chrome, tested both 6x and 20x

**Comment by Janpot** [comment]
That was on Chrome. This is firefox

https://github.com/user-attachments/assets/8a1466e1-a8fa-47d2-a5a8-f960d9ee2aad

Seeing the same on Safari, Edge and mobile chrome. But only when running locally, latest master. Not seeing it on the [deployed version](https://mui.com/material-ui/getting-started/templates/dashboard/). Maybe it's something poisoning the build locally 🤔

**Comment by JCQuintas** [comment]
Oddly enough, was seeing it happening when the CPU is not slowed down, but not after I open the `devtools`. Turns out the devtools made it withing the width range where the issue doesn't appear. 

Sweet spot is between ~912 - ~1590 where no shuflling happens 

I would assume this specific issue is due to some layout changes after css/js is loaded?

**Comment by Janpot** [comment]
I reduced the test fixture down to:

```tsx
import * as React from 'react';
import * as ReactDOMClient from 'react-dom/client';
import { Globals } from '@react-spring/web';
import Box from '@mui/material/Box';
import Grid from '@mui/material/Grid2';
import { SparkLineChart } from '@mui/x-charts/SparkLineChart';

Globals.assign({
  skipAnimation: true,
});

function StatCard() {
  return (
    <Grid size={6} sx={{ height: 50 }}>
      <SparkLineChart data={[1640, 220]} />
    </Grid>
  );
}

function App() {
  return (
    <Box sx={{ display: 'inline-block' }}>
      <Grid container spacing={2}>
        <StatCard />
        <StatCard />
        <StatCard />
      </Grid>
    </Box>
  );
}

const container = document.getElementById('react-root');
const children = <App />;
const reactRoot = ReactDOMClient.createRoot(container);
reactRoot.render(children);
```

It looks like there is some interaction going on between the top level `inline-block` in the [test viewer](https://github.com/mui/material-ui/blob/dcc17031aa37b251678c39628b15637ec758d6b4/test/regressions/TestViewer.js#L89), the `Grid2` and the `SparkLineChart`.

[codesandbox](https://codesandbox.io/p/sandbox/billowing-cloud-t6585q?file=%2Fsrc%2FDemo.tsx) behaves differently though.

I'm trying out removing this `inline-block` to see what it does, but even if that fixes the jitter, it still feels like there's some bug in the charts, I feel like it shouldn't be affected by this.

**Comment by JCQuintas** [comment]
@Janpot the issue here seems to be that the container doesn't have a proper width, which means the children has to "grow until they fit" into the container. 

One alternative would be to change the component that has the `display: 'inline-block'` to have a width as well. `<Box sx={{ display: 'inline-block', width: '100%' }}>`

I've created https://github.com/mui/mui-x/pull/14553 which should fix the issue in most common cases if the proper solution above is not possible.

**Comment by Janpot** [comment]
it's possible but it updates all the screenshots https://github.com/mui/material-ui/pull/43656
